tests/backup/backup_test.go

```diff
@@ -5322,13 +5554,54 @@ func TearDownBackupRestore(bkpNamespaces []string, restoreNamespaces []string) {
 	DeleteCluster(destinationClusterName, OrgID, ctx)
 	DeleteCluster(SourceClusterName, orgID, ctx)
 	// Need to add backup location UID for Delete Backup Location call
-	DeleteBackupLocation(backupLocationName, "", OrgID)
+	err = DeleteBackupLocation(backupLocationName, "", OrgID)
+	dash.VerifySafely(err, nil, fmt.Sprintf("Deleting  backup  location %s", backupLocationName))
 	DeleteCloudCredential(CredName, OrgID, CloudCredUID)
 }
+
 func getEnv(environmentVariable string, defaultValue string) string {
 	value, present := os.LookupEnv(environmentVariable)
 	if !present {
 		value = defaultValue
 	}
 	return value
 }
+
+// ShareBackupWithMultipleUsers shares backup with multiple users with different access levels
+// It returns a map with key as userAccessContext and value as backup shared
+func ShareBackupWithMultipleUsers(backupNames []string, users []string, ctx context.Context) (map[userAccessContext]string, error) {
+	log.InfoD("Sharing backups with users with different access level")
+	AccessUserBackupContext := make(map[userAccessContext]string)
+	var err error
+	var ctxNonAdmin context.Context
+	var access BackupAccess
+	for i, user := range users {
+		userIndex := i % 3
+		switch userIndex {
+		case 0:
+			access = ViewOnlyAccess
+			err = ShareBackup(backupNames[i], nil, []string{user}, access, ctx)
+		case 1:
+			access = RestoreAccess
+			err = ShareBackup(backupNames[i], nil, []string{user}, access, ctx)
+		case 2:
+			access = FullAccess
+			err = ShareBackup(backupNames[i], nil, []string{user}, access, ctx)
+		default:
+			access = ViewOnlyAccess
+			err = ShareBackup(backupNames[i], nil, []string{user}, access, ctx)
+		}
+		if err != nil {
+			err = fmt.Errorf("unable to share backup %s with user %s Error: %v", backupNames[i], user, err)
+			return AccessUserBackupContext, err
+		}
+		ctxNonAdmin, err = backup.GetNonAdminCtx(users[i], "Password1")
+		if err != nil {
+			err = fmt.Errorf("unable to share backup %s with user %s Error: %v", backupNames[i], user, err)
+			return AccessUserBackupContext, err
+		}
+		accessContextUser := userAccessContext{users[i], access, ctxNonAdmin}
+		AccessUserBackupContext[accessContextUser] = backupNames[i]
```

Comment:
Why is this in upper case? Do you want to `AccessUserBackupContext` to be used outside the package?

Reply:
changed